pod.go

```diff
@@ -525,12 +528,108 @@ func (p *Pod) createSetStates() error {
 	return nil
 }
 
+type assetType string
+
+const (
+	kernelAsset assetType = "kernel"
+	imageAsset            = "image"
+)
+
+func validateAsset(path string, t assetType) bool {
+	if !filepath.IsAbs(path) {
+		return false
+	}
+
+	// TODO Check for asset type
+	switch t {
+	case kernelAsset:
+		return true
+	case imageAsset:
+		return true
+	}
+
+	return true
+}
+
+func selectAssetFromAnnotation(podConfig *PodConfig, pathAnnotation, hashAnnotation string) (string, error) {
+	assetPath, ok := podConfig.Annotations[pathAnnotation]
+	if !ok {
+		return "", nil
+	}
+
+	if !filepath.IsAbs(assetPath) {
```

Comment:
This implicitly handles the case where `assetPath == ""`, but I wonder if this should be an explicit check (`if !ok || assetPath == ""` or something?). Same comment for `assetHashString`.